src/main/thrift/parquet.thrift

```diff
@@ -373,6 +408,74 @@ struct JsonType {
 struct BsonType {
 }
 
+/**
+ * Phyiscal type and encoding for the geometry type.
+ */
+enum GeometryEncoding {
+  /**
+   * Allowed for phyiscal type: BYTE_ARRAY.
+   *
+   * Well-known binary (WKB) representations of geometries. It supports 2D or
+   * 3D geometries of the standard geometry types (Point, LineString, Polygon,
+   * MultiPoint, MultiLineString, MultiPolygon, and GeometryCollection). This
+   * is the preferred option for maximum portability.
+   *
+   * This encoding enables GeometryStatistics to be set in the column chunk
+   * and page index.
+   */
+  WKB = 0;
+
+  /**
+   * Encodings from POINT to MULTIPOLYGON below are specialized for single
+   * geometry type and inspired by GeoArrow (https://geoarrow.org/format.html)
+   * native encodings. It uses the separated (struct) representation of
+   * coordinates for single-geometry type encodings because this encoding
+   * results in useful column statistics when row groups and/or files contain
+   * related features.
+   *
+   * WARNING: GeometryStatistics cannot be enabled for these encodings because
+   * only leaf columns can have column statistics and page index.
```

Comment:
I think WKB = 0 is pretty clear, but the GeoArrow ones are not so clear because they dictate that entire type must be one Geometry subtype (ie all Points, or all Lines, etc)?  And so it can be limiting for users.

Maybe as you already saw, but we were chatting with @jiayuasu on the Iceberg proposal , and they  had experimented with encodings like [GeoLake native](https://wherobots.notion.site/EXT-Geometry-encoding-type-and-expression-in-Iceberg-193f84ff42b44d2db326dc43f753598f#89e7f7a23e28437985db0ecc724f6127), that is a native encoding but re-uses same schema to represent all Geometry subtypes?  

As it may be a longer debate, may be we can delay adding non-WKB ones in this first pass?

Reply:
The current PR tries to add maximum support as far as I can see (including the native encoding type and various statistics) to seek opinions. I agree that we need to remove controversial/unclear things off in the final PR.

Reply:
Yes, as we discussed before, the current native encoding in GeoParquet (borrowed from GeoArrow) does not allow mixed types of geometries in the same column. In addition, its current design makes it hard for Iceberg to adopt. But I chatted with @paleolimbot last week, and GeoArrow community is willing to make necessary changes to make it work.

However, the main reason of having the native encoding in GeoParquet is to compute min/max statistics. If Parquet allows bbox / s2 / h3 as the native statistics, the native encoding does not seem to be necessary. This will greatly increase the adoption of Parquet / GeoParquet.

So, given the uncertainty of this topic, maybe we can remove this in the final PR.